cirq/vis/heatmap.py

```diff
@@ -184,86 +239,31 @@ def _plot_colorbar(
 
     def _write_annotations(self, ax: plt.Axes) -> None:
         """Writes annotations to the center of cells. Internal."""
-        for row, col in self.value_map.keys():
+        for target in self.value_map.keys():
+            row, col = self._target_to_coordinate(target)
             annotation = self.annot_map.get((row, col), '')
             if not annotation:
                 continue
-            text_color = (
-                'black' if self.value_map[(row, col)][0] > (self.vmax + self.vmin) / 2 else 'white'
+            medium_brightness = (
+                (self.vmax + self.vmin) / 2
+                if (self.vmax is not None and self.vmin is not None)
+                else 0.5
```

Comment:
It's reasonable to assume the value falls between 0 and 1 when self.vmax and self.vmin are not give, but what if it falls outside the range?